### PR TITLE
Add ARM-specific syscalls

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

**Comment by tbu-** [comment]
Needed for my mutex implementation.

**Comment by tbu-** [comment]
Mhmh, it seems that `arch/arm/tools/syscall.tbl` does exist on newer versions of the Linux source tree, so now I'm going to try to fix up the script instead.

**Comment by tbu-** [comment]
@homunkulus try

@japaric Should be fine to merge now; implementation to get the syscalls is a bit hacky.

File: tools/nr_from_src.py [hunk 1]
```diff
@@ -56,15 +65,18 @@ def load_headers(names: Iterable[str], arch: str, extra: str = ''):
...
-        subprocess.check_output(['git', '--no-pager', 'grep', '__NR_'], cwd=linux_path)
+    RE_SYSCALL_NR=re.compile(r'\b__([A-Z]+_)?NR_([a-z0-9_]+)\b')
+    names = set(x.groups() for x in RE_SYSCALL_NR.finditer(
+        subprocess.check_output(['git', '--no-pager', 'grep', r'\<__\([A-Z]\+_\)\?NR_'], cwd=linux_path)
             .decode('utf-8')))
     if len(names) < 380:
         print("didn't find anywhere near enough syscalls; hack must have failed")
+        subprocess.check_call(['git', '--no-pager', 'grep', r'\<__\([A-Z]\+_\)\?NR_'], cwd=linux_path)
         sys.exit(1)
+    ARM_NAMES = ["breakpoint", "cacheflush", "usr26", "usr32", "set_tls"]
```
Comment (japaric) [inline tools/nr_from_src.py]: hmm, this variable seems to be unused?

File: tools/nr_from_src.py [hunk 2]
```diff
@@ -22,32 +24,39 @@ def load_table(path: str, arches: Set[str]):
...
-        f.flush()
-        lines = subprocess.check_output(['gcc', '-nostdinc',
-            '-I', '{}/arch/{}/include/uapi'.format(linux_path, arch),
-            '-I', '{}/include'.format(linux_path),
-            '-P', # don't include line number markers, which make the output annoying to parse
-            '-E', # only preprocess, don't compile
-            f.name]).decode('utf-8').split('\n')
+        with tempfile.TemporaryDirectory() as temp_include_dir:
+            os.mkdir('{}/asm'.format(temp_include_dir))
+            with open('{}/asm/unistd-eabi.h'.format(temp_include_dir), 'w'):
```
Comment (japaric) [inline tools/nr_from_src.py]: is this like `touch unistd-eabi.h`? Could you add a comment about why this is needed?

**Comment by tbu-** [comment]
Fixed the issues mentioned in the comments.

**Comment by japaric** [comment]
Thanks @tbu-.

@homunkulus r+

**Comment by homunkulus** [comment]
:pushpin: Commit 07fb52d has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit 07fb52d with merge 07fb52d...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/syscall.rs/builds/229449596?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 07fb52dd950ff32d91d5436755f084e4223bfb81 to master...

**Comment by tbu-** [comment]
Could you release a new crates.io version with this?

**Comment by japaric** [comment]
@tbu- done

**Pull request merged**